### PR TITLE
Add ships

### PR DESCRIPTION
**Pull request by lrharris215**
Created Ship and ShipSection models.

### CODE REVIEW

File: src/main/java/com/battleship/backend/controllers/ShipController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public @ResponseBody
+    Ship getShip(){
+        Ship ship = new Ship("Test Ship", 4);
+        ShipSection[] shipSections = ship.getShipSections();
+        for(ShipSection shipSection : shipSections){
+            shipSection.receiveHit();
+        }
+        ship.checkIsSunk();
+        return ship;
+    }
```
Comment (kudukacar) [inline src/main/java/com/battleship/backend/controllers/ShipController.java:22]: As discussed, we don't need this controller now.

File: src/test/java/com/battleship/backend/models/ShipTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    void testCheckIsSunkReturnsFalseIfNoSectionsAreHit(){
+        assertFalse(testShip.checkIsSunk());
+    }
+    @Test
+    void testCheckIsSunkReturnsFalseIfOnly1SectionIsHit(){
+        testShip.getShipSections()[0].receiveHit();
+
+        assertFalse(testShip.checkIsSunk());
+    }
+
```
Comment (kudukacar) [inline src/test/java/com/battleship/backend/models/ShipTest.java:44]: You don't need to test setters or getters

File: src/main/java/com/battleship/backend/controllers/ShipController.java (hunk 1 above)
Comment (lrharris215) [reply src/main/java/com/battleship/backend/controllers/ShipController.java]: Removed!

File: src/test/java/com/battleship/backend/models/ShipTest.java (hunk 2 above)
Comment (lrharris215) [reply src/test/java/com/battleship/backend/models/ShipTest.java:44]: Removed!

**Review by kudukacar — APPROVED** [review]
Looks good.

**Pull request merged**